pandas/plotting/_matplotlib/style.py

```diff
@@ -94,3 +74,41 @@ def _maybe_valid_colors(colors):
         colors += colors[:mod]
 
     return colors
+
+
+def _is_cn_color(color: str) -> bool:
+    """Check if color string is CN color, like 'C0', 'C1', etc."""
+    cn_colors = ["C" + str(x) for x in range(10)]
+    return bool(color in cn_colors)
+
+
+def _is_single_color(color: str) -> bool:
+    """Check if ``color`` is a single color.
+
+    Examples of single colors:
+        - 'r'
+        - 'g'
+        - 'red'
+        - 'green'
+        - 'C3'
+
+    Parameters
+    ----------
+    color : string
+        Color string.
+
+    Returns
+    -------
+    bool
+        True if ``color`` looks like a valid color.
+        False otherwise.
+    """
+    conv = matplotlib.colors.ColorConverter()
+    if _is_cn_color(color):
```

Comment:
`to_rgba` handles the `"Cn"` style, so why is this separate check needed?

Reply:
Correct! It turns out that we can make it even cleaner.